iModelCore/ecobjects/src/ECSchema.cpp

```diff
@@ -3211,6 +3211,18 @@ ECSchemaPtr SearchPathSchemaFileLocater::_LocateSchema(SchemaKeyR key, SchemaMat
     return schemaOut;
     }
 
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+ECSchemaPtr StringSchemaLocater::_LocateSchema(SchemaKeyR key, SchemaMatchType matchType, ECSchemaReadContextR schemaContext)
+    {
+    ECSchemaPtr schema;
+    auto schemaXml = m_schemaXml.find(key.m_schemaName);
+    if (schemaXml != m_schemaXml.end())
+        ECSchema::ReadFromXmlString(schema, schemaXml->second.c_str(), schemaContext);

```

Comment:
This method returns a status code. Now that this is no longer test but production code, let's evaluate the return code and log an error message if it's not success.

Reply:
Error messages are already logged by the ECSchema::ReadFromXmlString() method if the schema XML string cannot be deserialized or a duplicate schema is detected.